lib/java/src/org/apache/thrift/async/TAsyncMethodCall.java

```diff
@@ -225,8 +231,13 @@ private void cleanUpAndFireCallback(SelectionKey key) {
     key.interestOps(0);
     // this ensures that the TAsyncMethod instance doesn't hang around
     key.attach(null);
```

Comment:
NB: I still don;t fully understand this code, but you've got me reading it more closely.  I'll report back when I fully grok what's going on here in the pre-existing code.